pkg/platform/runtime/setup/buildlog/buildlog.go

```diff
@@ -58,7 +58,9 @@ func New(artifactMap map[artifact.ArtifactID]artifact.ArtifactRecipe, conn Build
 	errCh := make(chan error)
 
 	total := len(artifactMap)
-	events.BuildStarting(total)
+	// We set the total to one extra artifact, as we always have to wait for
+	// the terminal artifact to finish before the build is actually done.
+	events.BuildStarting(total + 1)
```

Comment:
This feels error prone, is there no way we can get an accurate count via the API, without making assumptions?

Reply:
That is a very good point!

The issue that I wanted to fix was: I encountered a build, where the installation was done, the build progress bar *indicated* that it was done, but the terminal artifact was still building, and so the buildlog instance was still waiting for it - the installation just halted.

I can think of two better ways to handle this:
1. We use the total number of artifacts that need to be built from the head-chef's `BuildStatusResponse` (not from the Recipe as we do now) which has all the artifact IDs that are actually being built.  We would not be able to send the `artifactName` with every build event anymore, but our current progressbar implementation is not doing anything with that information.
2. We update the progressbar such that it prints "Waiting for build to finish" (or something like that) after all expected artifacts have been built, but before the `BuildFinished` event has been received.

I personally favor Option 1, as it seems more honest, and is also much simpler to implement. Also: If we really ever want to display the names of currently building artifacts, there are other ways to achieve that.

But I should probably take this part out of this PR, and file a follow-up story...?

Reply:
@Naatan ^

Reply:
Yeah let's take it out. Or maybe address it with option 2 for now and we'll iterate with option 1 later. I do think we should show the names of the artifacts that are building, but that can be part of the iteration.

Reply:
Just to clarify:  Currently our progressbar is *not* printing the names of artifacts that are *building*; we do that only for the *download*, *unpack* and *install* steps.

I'll take it out, because Option 2 is actually difficult to implement with the `mpb.Progress` package.

Reply:
https://www.pivotaltracker.com/story/show/177631597